src/garage/envs/env_spec.py

```diff
@@ -11,5 +13,44 @@ class EnvSpec:
     """
 
     def __init__(self, observation_space, action_space):
-        self.observation_space = observation_space
-        self.action_space = action_space
+        super().__init__(action_space, observation_space)
+
+    @property
+    def action_space(self):
+        """Get action space.
+
+        Returns:
+            akro.Space: Action space of the env.
+
+        """
+        return self.input_space
+
+    @property
+    def observation_space(self):
+        """Get observation space of the env.
+
+        Returns:
+            akro.Space: Observation space.
+
+        """
+        return self.output_space
+
+    @action_space.setter
```

Comment:
please do not have setters. this omission was intentional -- this is supposed to be immutable. was there any need for this somewhere?

Reply:
A lot of existing tests set `action_space` and `observation_space`, so they are failing with setters.

Reply:
ah -- please instead file an issue about making these immutable